src/services/Randomizer.hxx

```diff
@@ -3,31 +3,32 @@
 
 #include <random>
 #include <iterator>
+#include "../util/Singleton.hxx"
 
-#include "../GameService.hxx"
-#include "LOG.hxx"
-
-using RNG = std::mt19937;
-class Randomizer : public GameService
+class Randomizer : public Singleton<Randomizer>
 {
-  thread_local static RNG generator;
+  static thread_local std::mt19937 generator;
 
-  template <typename IntType> using UniformInt = std::uniform_int_distribution<IntType>;
+  template <typename IntType> using UniformIntDist = std::uniform_int_distribution<IntType>;
 
 public:
-  Randomizer(GameService::ServiceTuple &context) : GameService(context) {}
+  Randomizer() = default;
+
+  template <typename Iterator> void shuffle(Iterator begin, Iterator end) { std::shuffle(begin, end, generator); }
 
   /**
    * @returns a random iterator between [begin, end)
```

Comment:
Conventionally the returns command is at the end/after the params command, but I don't think that impacts the generated documentation and it's not a big deal, just something that popped out at me when reading this as code.

Reply:
Good point, I fixed it.